extensions/wikia/Rail/RailController.class.php

```diff
@@ -8,139 +8,163 @@ class RailController extends WikiaController {
 	}
 
 	/**
 	 * Get lazy right rail modules
 	 */
 	protected function getLazyRail() {
-		wfProfileIn(__METHOD__);
-		global $wgUseSiteJs, $wgAllowUserJs, $wgTitle, $wgAllInOne;
-		$title = Title::newFromText($this->request->getVal('articleTitle', null), $this->request->getInt('namespace', null));
+		global $wgAllInOne;
+
+		$context = RequestContext::getMain();
 
-		if ($title instanceof Title) {
+		$title = Title::newFromText(
+			$this->request->getVal( 'articleTitle', null ),
+			$this->request->getInt( 'namespace', null )
+		);
+
+		if ( !( $title instanceof Title ) ) {
+			return;
+		}
+
+		$wrapper = new Wikia\Util\GlobalStateWrapper( [
+			// Do not load user and site jses as they are already loaded and can break page
+			'wgAllowUserJs' => false,
+			'wgUseSiteJs' => false,
 			// override original wgTitle from title given in parameters
 			// we cannot use wgTitle that is created on by API because it's broken on wikis without '/wiki' in URL
 			// https://wikia-inc.atlassian.net/browse/BAC-906
-			$oldWgTitle = $wgTitle;
-			$wgTitle = $title;
-			$assetManager = AssetsManager::getInstance();
-			$railModules = $this->filterModules((new BodyController)->getRailModuleList(), self::FILTER_LAZY_MODULES);
-			$this->railLazyContent = '';
-			krsort($railModules);
-			foreach ($railModules as $railModule) {
-				$this->railLazyContent .= $this->app->renderView(
-					$railModule[0], /* Controller */
-					$railModule[1], /* Method */
-					$railModule[2] /* array of params */
-				);
-			}
+			'wgTitle' => $title
+		] );
 
-			$this->railLazyContent .= Html::element('div', ['id' => 'WikiaAdInContentPlaceHolder']);
+		$assetManager = AssetsManager::getInstance();
 
-			$this->css = $sassFiles = [];
-			foreach (array_keys($this->app->wg->Out->styles) as $style) {
-				if ($wgAllInOne && $assetManager->isSassUrl($style)) {
-					$sassFiles[] = $style;
-				} else {
-					$this->css[] = $style;
-				}
-			}
+		$railModuleListRaw = $wrapper->wrap( function () use ( $context ) {
+			return ( new BodyController )->getRailModuleList();
+		} );
 
-			if (!empty($sassFiles)) {
-				$excludeScss = (array) $this->getRequest()->getVal('excludeScss', []);
-				$sassFilePath = (array) $assetManager->getSassFilePath($sassFiles);
-				$includeScss = array_diff($sassFilePath, $excludeScss);
+		$railModules = $this->filterModules(
+			$railModuleListRaw,
+			self::FILTER_LAZY_MODULES
+		);
+		$this->railLazyContent = '';
 
-				// SUS-771: Log any duplicate CSS that rail modules try to load but are already loaded by Oasis skin
-				$duplicateScss = array_intersect( $sassFilePath, $excludeScss );
-				if ( count( $duplicateScss ) ) {
-					Wikia\Logger\WikiaLogger::instance()->info( 'SUS-771', [
-						'styles' => json_encode( $duplicateScss )
-					] );
-				}
+		krsort( $railModules );
 
-				if (!empty($includeScss)) {
-					$this->css[] = $assetManager->getSassesUrl($includeScss);
-				}
-			}
+		// TODO XW-2760 remove after experiment is done
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)